packages/libraries/client/src/client.ts

```diff
@@ -9,25 +9,34 @@ import { version } from './version.js';
 
 export function createHive(options: HivePluginOptions): HiveClient {
   const logger = options?.agent?.logger ?? console;
+  let enabled = options.enabled ?? true;
 
-  if (!options.enabled) {
+  if (enabled === false) {
     logIf(options.debug === true, '[hive] is not enabled.', logger.info);
   }
 
-  if (!options.token && options.enabled) {
-    options.enabled = false;
+  if (!options.token && enabled) {
+    enabled = false;
     logger.info('[hive] Missing token, disabling.');
   }
 
-  const usage = createUsage(options);
-  const schemaReporter = createReporting(options);
-  const operationsStore = createOperationsStore(options);
+  const mergedOptions: HivePluginOptions = { ...options, enabled } as HivePluginOptions;
+
+  const usage = createUsage(mergedOptions);
+  const schemaReporter = createReporting(mergedOptions);
+  const operationsStore = createOperationsStore(mergedOptions);
 
   function reportSchema({ schema }: { schema: GraphQLSchema }) {
-    void schemaReporter.report({ schema });
+    if (enabled === false) {
+      return;
+    }
```

Comment:
```suggestion
```

The `createReporting` itself reacts to `enabled`.
The reason why it's there and not here is that the `HiveClient` expects a specific return type from these functions and it's a bit easier to have everything next to each other, types, real logic and noop.